### DISCUSSION TITLE
-cdn -json should return true or false

### DISCUSSION THREAD

**Discussion by brenocss** [projectdiscovery/httpx#497] [Ideas] [answered]
### httpx version:
v1.1.5

### Current Behavior:

```bash
echo 'google.com' | httpx -cdn -json | jq .cdn
```
`null`

### Expected Behavior:

```bash
echo 'google.com' | httpx -cdn -json | jq .cdn
```
`false`

### Use case
```bash
echo 'google.com
hackerone.com' | httpx -json >> log.json

echo 'google.com
hackerone.com' | httpx -cdn -json >> log.json
```
```bash
# Get all hosts's that is from any cdn
cat log.json | jq 'select(.cdn==true).url'
```

```bash
# Get all hosts's that isn't from any cdn
cat log.json | jq 'select(.cdn==false).url'
# cat log.json | jq 'select(.cdn!=true).url' doesn't work because it returns null with or without the -cdn flag, getting invalid results 
```

**Comment by ehsandeep** [comment, marked as answer]
@brenocss this feature will be updated with https://github.com/projectdiscovery/httpx/issues/492, so instead of returning true/false, it will return the name of the cdn.

**Reply by brenocss** [reply]
The best of all worlds would be to have both fields that says if belongs to a cdn and the name of that cdn
```bash
echo 'hackerone.com' | httpx -cdn -cdn-title
https://api.hackerone.com [cloudflare]
```
```json
{
  "cdn":true||false,
  "cdn_title": "cloudflare"
}
```

**Reply by ehsandeep** [reply]
@brenocss `cat log.json | jq -r 'select(.cdn!=true).url'` this works with latest changes in the dev, and returning `false` may not be a good idea because the [cdncheck](https://github.com/projectdiscovery/cdncheck) library only returns `true` for CIDR ranges that are known to us but not all.
